### ISSUE TITLE
reveal message does not work for first time

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#20306]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. from userA, send a message to userB
2. from userB , flag as offensive message
3. from userA, click on edit message option and edit message
4. click on reveal message button
5. click on :heavy_check_mark: button
## Expected Result: 
it should reveal message
## Actual Result: 
should not reveal message

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.24-4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/e2901a13-3257-479d-996a-6440f330d662

https://github.com/Expensify/App/assets/43996225/b6b6143e-1805-4cd7-b607-5484c658aeab

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1685604959779469
 
[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kadiealexander** [comment]
Reproduced! 

https://github.com/Expensify/App/assets/59587260/bcd42f92-f4e9-404a-bddc-7bd2b782379c

This is an odd flow though, since the message is technically already "revealed" during the edit, I wonder if the button should be removed when the edit field is open. cc @thienlnam @dangrous for advice on expected behaviour here.

**Comment by kadiealexander** [comment]
Updated the post to reflect this, thanks team! 

@hungvu193, could you please adjust your proposal to ensure the "Reveal message" button is hidden while the editor is open?

**Comment by kadiealexander** [comment]
Thank you!! I'll leave these for @aimane-chnaif to review :)

**Comment by kadiealexander** [comment]
>The purpose of that button is just to see what message was hidden. So I think it should be temporary and no need to save longer.
> The only odd thing is when 👎, it's reverted.
> BTW that's just my suggestion and still open for any better suggestions. cc: @robertjchen @kadiealexander @thienlnam
> 
> Option 1: return back to hidden whenever page refresh
> Option 2: give time interval so after 3s, hide automatically
> Option 3: store local-only in Onyx, so when logout and re-login, it's lost and returns back. And never reflected on another device even with same account.
> Option 4: store both local and server per account basis. I am 👎 on this, maybe others think same.

I think option one (show until the tab is refreshed) makes the most sense here and is in line with how our product behaves. Keen for @thienlnam and @robertjchen's opinions here before proceeding though.

**Comment by thienlnam** [comment]
Agreed, no need to overcomplicate the solution - let's keep it simple and go with option 1

**Comment by robertjchen** [comment]
Yes, I am in agreement- we should revert the message as hidden upon refresh. In that case, it looks like we should proceed with @dukenv0307 's proposal

**Comment by kadiealexander** [comment]
@anmurali I'm going OOO so I've reassigned this to ensure everyone gets paid on time. I've sent out most of the offers which you can find [here](https://www.upwork.com/ab/applicants/1666657004353064960/hired), and @aimane-chnaif and @dukenv0307 are due an #urgency bonus for their quick work ([here](https://github.com/Expensify/App/issues/20306#issuecomment-1590515132)). Thanks so much for your help!

**Comment by anmurali** [comment]
Paid everyone but @dukenv0307 
Pinged him [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1687800453097169) to get his Upwork profile so I can hire and pay him

**Comment by robertjchen** [comment]
Looks like it was accepted here, closing this out for now: https://expensify.slack.com/archives/C01GTK53T8Q/p1687866690845559?thread_ts=1687800453.097169&cid=C01GTK53T8Q
